pkg/registry/generic/etcd/etcd.go

```diff
@@ -271,10 +272,21 @@ func (e *Etcd) Update(ctx api.Context, obj runtime.Object) (runtime.Object, bool
 	if err != nil {
 		return nil, false, err
 	}
+	// If the object specified by the user does not have a resource version,
```

Comment:
I would want this feature to be optional on a type by type basis - opt in.  We have a few objects where I'd want to discourage this pattern.